### PR TITLE
feat: Adds "Combat Locations" (Raids/Dungeons/Caves/Grind Spots/Shrines) to the Map

### PR DESCRIPTION
**Pull request by san7890**
Now Ready For Review:

image

As on the PR Title, adds these Points of Interest to the in-game map.

### CODE REVIEW

**Comment by magicus** [comment]
Is the icon art from Legacy or newly designed?

**Comment by kristofbolyai** [comment]
> Is the icon art from Legacy or newly designed?

I think all of them is reused. Raids didn't have a texture, and corrupted dungeon icon was reused here for that, as it is not used for corrupted dungeons.

**Comment by magicus** [comment]
Ah, I see. That makes sense.

**Comment by kristofbolyai** [comment]
Hey! @san7890 as we agreed on discord, I fixed up your PR to parse the combat locations.

**Comment by san7890** [comment]
@kristofbolyai i'm not able to get it to work on my local, it's possible i broke something by updating the JSONs?

regardless, thank you very much!

**Comment by kristofbolyai** [comment]
> @kristofbolyai i'm not able to get it to work on my local, it's possible i broke something by updating the JSONs?
> 
> 
> 
> regardless, thank you very much!

Maybe, ill check later.

**Comment by kristofbolyai** [comment]
> @kristofbolyai i'm not able to get it to work on my local, it's possible i broke something by updating the JSONs?
> 
> 
> 
> regardless, thank you very much!

My refactor broke serialization i think

**Comment by kristofbolyai** [comment]
@san7890 I tested it, and apart from merge issues, it seems to work? Maybe try now?

File: common/src/main/java/com/wynntils/wynn/model/map/poi/CombatKind.java [hunk 1]
```diff
@@ -13,7 +13,9 @@ public enum CombatKind {
     DUNGEONS("Dungeons", Texture.DUNGEON_ENTRANCE),
     GRIND_SPOTS("Grind Spots", Texture.GRIND_SPOT),
     RAIDS("Raids", Texture.RAID_ENTRANCE),
-    RUNE_SHRINES("Rune Shrines", Texture.RAID_ENTRANCE);
+    // TODO: We only have sprites for Uth Shrines, so those are the only ones that we have
+    // location markers for. Split up the category when more textures are available.
```
Comment (magicus) [inline common/src/main/java/com/wynntils/wynn/model/map/poi/CombatKind.java]: If you are not happy with Uth shrine markers representing all shrines, then I'd rather say we need to update the texture to look more generic. I think we should avoid going down the route with having too specific markers.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/wynn/model/map/poi/CombatKind.java]: https://static.wikia.nocookie.net/wynncraft_gamepedia_en/images/2/20/WynnIconCSS.png/revision/latest?cb=20210928042006?v=20221206024249 has all runes as textures. You just need to crop them. I hope it is not too low quality. Although, if you check, our texture is a combination of UTH and AZ (color and shape). So it is "generic".

Comment (san7890) [reply common/src/main/java/com/wynntils/wynn/model/map/poi/CombatKind.java]: Oh, that's interesting

Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/map/poi/CombatKind.java]: To clarify the genericness, maybe we should design one with the same look-and-feel, but a furthark rune R?

Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/map/poi/CombatKind.java]: Here's a design or what I mean: https://imgur.com/a/jVs8ZYY:

img

**Comment by san7890** [comment]
@kristofbolyai Yeah, unfortunately still doesn't work for some odd reason. I cleared the `apicache` folder yet I'm still getting nothing:

image

Could this qodana issue have something to do with it?

image

I'm using Fabric. Apologies if I'm doing something wrong somehow.

**Comment by kristofbolyai** [comment]
> @kristofbolyai Yeah, unfortunately still doesn't work for some odd reason. I cleared the `apicache` folder yet I'm still getting nothing:
> 
> 
> 
> image
> 
> 
> 
> Could this qodana issue have something to do with it?
> 
> 
> 
> image
> 
> 
> 
> I'm using Fabric. Apologies if I'm doing something wrong somehow.

You have to render it. I only did parsing. Check renderPoi code for both minimap and main map. It is quite straightforward to get this rendered after that. Let me know if you need help :)

**Review by magicus — DISMISSED** [review]
I think this looks good now...

**Comment by kristofbolyai** [comment]
> Now Ready For Review:
> 
> image
> 
> As on the PR Title, adds these Points of Interest to the in-game map.

These icons look really weird. I want to check this out locally.

File: common/src/main/java/com/wynntils/features/user/map/MinimapFeature.java [hunk 2]
```diff
@@ -258,6 +258,8 @@ private void renderPois(
                     .toList();
             poisToRender.addAll(playerPois);
 
+            poisToRender.addAll(new ArrayList<>(MapModel.getCombatPois()));
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/map/MinimapFeature.java]:
```suggestion
            poisToRender.addAll(MapModel.getCombatPois());
```

I'm confident this works

**Comment by kristofbolyai** [comment]
Aha! One of the texture sizes was wrong! I fixed it.

**Review by magicus — APPROVED** [review]

**Pull request merged**